lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php

```diff
@@ -41,15 +41,13 @@ class XmlDriver extends FileDriver
 {
     public const DEFAULT_FILE_EXTENSION = '.dcm.xml';
 
-    /** @var bool */
+    /** @var bool|null */
     private $isXsdValidationEnabled;
 
     /**
      * {@inheritDoc}
-     *
-     * @param bool $isXsdValidationEnabled
      */
-    public function __construct($locator, $fileExtension = self::DEFAULT_FILE_EXTENSION, $isXsdValidationEnabled = false)
+    public function __construct($locator, $fileExtension = self::DEFAULT_FILE_EXTENSION, ?bool $isXsdValidationEnabled = false)
```

Comment:
Why allow null here? I'm on my phone, AFK for several days, so I can hardly navigate and take a look at the failing build, but this seems wrong anyway.

Reply:
My main issue is in `ORMSetup`:

```php

    public static function createXMLMetadataConfiguration(
        array $paths,
        bool $isDevMode = false,
        ?string $proxyDir = null,
        ?CacheItemPoolInterface $cache = null,
        ?bool $isXsdValidationEnabled = false
    ): Configuration {
        $config = self::createConfiguration($isDevMode, $proxyDir, $cache);
        $config->setMetadataDriverImpl(new XmlDriver($paths, XmlDriver::DEFAULT_FILE_EXTENSION, $isXsdValidationEnabled));

        return $config;
    }
```

If I don't make `$isXsdValidation` nullable I have to put it before the other nullable arguments which could cause BC break. I also hesitated to add a check here and transform the variable to false if null, but I wondered if it really was `ORMSetup` responsibility.

Let me know what you think ! BTW I'm also supposed to be AFK for a few days, so no hurry for this branch :).

Reply:
https://3v4l.org/da7c4 it seems to work? here, i put an arg with a default value after a nullable one, it compiles

Reply:
I modified it and run all the Quality and unit tests, seems to work. Maybe it was PHPStan which complained about it in some other projects :thinking:

Reply:
Or maybe you ran in an issue with a null able argument that was after an optional one? That definitely wouldn't work.